### PR TITLE
[PasswordHasher][Security] Remove legacy password encoders

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Time to clean :)

### CODE REVIEW

**Comment by chalasr** [comment]
Builds failures unrelated/false positives.  Ready

File: src/Symfony/Component/Security/Core/Tests/Validator/Constraints/UserPasswordValidatorTest.php [hunk 1]
```diff
@@ -36,26 +36,26 @@ abstract class UserPasswordValidatorTest extends ConstraintValidatorTestCase
     protected $tokenStorage;
 
     /**
-     * @var PasswordEncoderInterface
+     * @var PasswordHasherInterface
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Core/Tests/Validator/Constraints/UserPasswordValidatorTest.php]: shouldn't we do these changes on a lower branch first?

**Comment by chalasr** [comment]
Rebased.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**